test/e2e/windows/gmsa.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package windows
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+)
+
+var _ = SIGDescribe("[Feature:Windows] [Feature:WindowsGMSA] GMSA [Slow]", func() {
+	f := framework.NewDefaultFramework("gmsa-test-windows")
+
+	ginkgo.Describe("kubelet GMSA support", func() {
+		ginkgo.Context("when creating a pod with correct GMSA credential specs", func() {
+			ginkgo.It("passes the credential specs down to the Pod's containers", func() {
+				defer ginkgo.GinkgoRecover()
+
+				podName := "with-correct-gmsa-annotations"
+
+				container1Name := "container1"
+				podDomain := "acme.com"
+
+				container2Name := "container2"
+				container2Domain := "contoso.org"
+
+				containers := make([]corev1.Container, 2)
+				for i, name := range []string{container1Name, container2Name} {
+					containers[i] = corev1.Container{
+						Name:    name,
+						Image:   imageutils.GetPauseImageName(),
+						Command: []string{"powershell"},
+						Args:    []string{"while ($true) { Start-Sleep -Seconds 1 }"},
```

Comment:
Well, the command and args are unnecessary. It is a pause image after all, it is supposed to do that anyways.

Reply:
Good catch, thanks. Changed.

Sorry I forgot changing this as part of 9eac1bebd024ccf510ab126c08d7dfacd1e651d0 (was using image `mcr.microsoft.com/powershell` before).